wrappers/tempo_client.py

```diff
@@ -0,0 +1,110 @@
+import json
+import requests
+
+from dateutil import parser
+
+from .multi_log import MultiLog
+
+
+class TempoClient(object):
+    def __init__(self, config):
+        self.jira_url = config.get('jira_url')
+        self.tempo_api = config.get('tempo_api')
+        self.tempo_user = config.get('tempo_user')
+        self.tempo_password = config.get('tempo_password')
+        self.session = self.init_session()
+
+    def init_session(self):
+        session = requests.Session()
+        session.auth = (self.tempo_user, self.tempo_password)
+
+        api = self.jira_url + 'rest/api/2'
+        resp = session.get('%s/myself' % api)
+        if resp.ok:
+            return session
+        elif resp.status_code == 401:
+            raise Exception("Error: Incorrect password or username.")
+        elif resp.status_code == 403:
+            raise Exception(
+                "Jira credentials seems to be correct, but this user does "
+                "not have permission to log in.\nTry to log in via browser, "
+                "maybe you need to answer a security question: %s" %
+                self.jira_url
+            )
+        else:
+            raise Exception(
+                "Something went wrong,"
+                " Jira gave %s status code." % resp.status_code
+            )
+
+    def get_worklogs(self, date_window):
+        params = {
+            'jira_url': self.jira_url,
+            'tempo_api': self.tempo_api,
+            'date_from': date_window.start.date().isoformat(),
+            'date_to': date_window.stop.date().isoformat(),
+            'tempo_user': self.tempo_user,
+        }
+
+        url = "{jira_url}{tempo_api}worklogs" \
+              "?username={tempo_user}&dateFrom={date_from}&dateTo={date_to}"
```

Comment:
So you didn't solved the pagination problem you blame on the other script? :confused:

Reply:
There is no pagination in the Tempo API